pystiche/papers/sanakoyeu_et_al_2018/loss.py

```diff
@@ -0,0 +1,293 @@
+from collections import OrderedDict
+from typing import Callable, Optional, Sequence, Tuple, Union
+
+import torch
+import torch.nn as nn
+from torch.nn.functional import binary_cross_entropy_with_logits
+
+import pystiche
+from pystiche.enc import Encoder
+from pystiche.loss.perceptual import PerceptualLoss
+from pystiche.ops import (
+    EncodingOperator,
+    EncodingRegularizationOperator,
+    MultiLayerEncodingOperator,
+)
+from pystiche.ops.comparison import MSEEncodingOperator
+from pystiche.papers.sanakoyeu_et_al_2018.modules import (
+    SanakoyeuEtAl2018Discriminator,
+    SanakoyeuEtAl2018DiscriminatorEncoder,
+    SanakoyeuEtAl2018TransformerBlock,
+)
+
+from .utils import ContentOperatorContainer
+
+
+class DiscriminatorEncodingOperator(EncodingRegularizationOperator):
+    def __init__(
+        self,
+        encoder: SanakoyeuEtAl2018DiscriminatorEncoder,
```

Comment:
Although we use it as `encoder` internally, we should name it `discriminator` on this interface. We could also include a "static link" as property from `discriminator` to `encoder` to make our code more readable.

Reply:
To clarify the previous explanation a bit more this should be a `SingleLayerEncoder`.